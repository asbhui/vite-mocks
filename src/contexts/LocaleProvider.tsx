import { I18nextProvider } from 'react-i18next';
import i18nLoader from '../services/locales';

export const LocaleProvider = ({ children }: { children: React.ReactNode }) => (
  <I18nextProvider i18n={i18nLoader}>{children}</I18nextProvider>
);
