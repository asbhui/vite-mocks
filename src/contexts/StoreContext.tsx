import { createContext, PropsWithChildren } from 'react';
import { rootStore } from '../stores/RootStore';

export const StoreContext = createContext<typeof rootStore | undefined>(undefined);

export const StoreProvider = ({ children }: PropsWithChildren) => (
  <StoreContext.Provider value={rootStore}>{children}</StoreContext.Provider>
);
