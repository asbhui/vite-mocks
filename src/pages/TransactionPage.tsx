import { observer } from 'mobx-react-lite';
import { InputForm } from '../components/mobxExample/InputForm';

import { useTranslation } from 'react-i18next';
import { Stack, Typography } from '@mui/material';
import { TransactionDisplay } from '../components/mobxExample/TransactionDisplay';

export const TransactionPage = observer(() => {
  const { t } = useTranslation();
  return (
    <Stack gap={2} alignItems={'center'}>
      <Typography variant="h2">{t('transactionPage.title')}</Typography>
      <Stack direction={'row'} gap={2} justifyContent={'flex-start'} alignItems={'flex-start'}>
        <InputForm />
        <TransactionDisplay />
      </Stack>
    </Stack>
  );
});
