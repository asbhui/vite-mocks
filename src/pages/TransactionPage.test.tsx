import { render } from '@testing-library/react';
import { TransactionPage } from './TransactionPage';
import { StoreProvider } from '../contexts/StoreContext';

describe('TransactionPage test', () => {
  it('should render transaction page', () => {
    const { container } = render(
      <StoreProvider>
        <TransactionPage />
      </StoreProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
