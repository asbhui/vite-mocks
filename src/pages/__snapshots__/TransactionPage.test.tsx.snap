// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TransactionPage test > should render transaction page 1`] = `
<div>
  <div>
    <h1>
      Transaction Lookup
    </h1>
    <form>
      Email:
      <input
        placeholder="Email"
        required=""
        type="email"
        value=""
      />
      <br />
      Transaction ID:
      <input
        placeholder="Transaction ID"
        required=""
        type="text"
        value=""
      />
      <br />
      <button
        type="submit"
      >
        Fetch Transaction
      </button>
    </form>
  </div>
</div>
`;
